spython/main/parse/parsers/base.py

```diff
@@ -0,0 +1,95 @@
+
+# Copyright (C) 2017-2019 Vanessa Sochat.
+
+# This Source Code Form is subject to the terms of the
+# Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+# with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+from spython.logger import bot
+from spython.utils import read_file
+from ..recipe import Recipe
+
+class ParserBase(object):
+    '''a parser Base is intended to provide helper functions for a parser,
+       namely to read lines in files, and otherwise interact with outputs.
+       Input should be some recipe (text file to describe a container build)
+       and output of parse() is a Recipe (spython.main.parse.recipe.Recipe)
+       object, which can be used to write to file, etc.
+    '''
+
+    lines = []
+
+    def __init__(self, filename=None, load=True):
+        '''a generic recipe parser holds the original file, and provides 
+           shared functions for interacting with files. If the subclass has
+           a parse function defined, we parse the filename
+
+           Parameters
+           ==========
+           recipe: the recipe file to parse.
+
+        '''
+        self.filename = filename
+        self._run_checks()
+        self.recipe = Recipe(self.filename)
+
+        if self.filename:
+
+            # Read in the raw lines of the file
+            self.lines = read_file(self.filename)
+
+            # If parsing function defined, parse the recipe
+            if load is True and hasattr(self, 'parse'):
+                self.parse()
```

Comment:
Create abstract method and simply call it

Reply:
Okay, I've never done this (using abc?) but I'll give it a try.